hrm-domain/hrm-service/src/data-pull-task/khp-data-pull-task/index.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import startOfDay from 'date-fns/startOfDay';
+import endOfDay from 'date-fns/endOfDay';
+import subHours from 'date-fns/subHours';
+import parse from 'date-fns/parse';
+import isValid from 'date-fns/isValid';
+
+import { pullCases } from './pull-cases';
+import { pullContacts } from './pull-contacts';
+
+const hasNoDateArgs = (args: any) =>
+  args['start-date'] === undefined && args['end-date'] === undefined;
+
+const getDateRangeForPast12Hours = () => {
+  const endDate = new Date();
+  const startDate = subHours(endDate, 12);
+
+  return { startDate, endDate };
+};
+
+const getDateRangeFromArgs = (args: any) => {
+  const startDateFromArgs = args['start-date'];
+  const endDateFromArgs = args['end-date'];
+
+  // Question: Should we deal with timezones explicitally?
+  const givenStartDate = parse(startDateFromArgs, 'yyyy-MM-dd', new Date());
```

Comment:
We could just say that the args can be any valid ISO format (i.e. use `parseISO` from date-fns), this would allow users to specify dates only, date/times and decide whether to explicitly specify a time zone themselves